### PR TITLE
bring in ACK runtime v0.12.0

### PR DESCRIPTION
**Pull request by jaypipes**
Regenerates controller with ACK runtime v0.12.0.

In the process of running e2e tests after the regeneration, I ran into
an error related to the e2e tests not ensuring that a service-linked
role was associated with the AWS account for the private VPC access
Elasticsearch Domain. I have added a bootstrap/cleanup for the necessary
service-linked role.

The IAM role associated with the controller service account used in
testing needs to have the ability to create and delete service-linked
roles. The IAM role used for testing in the ES team's CI account does
indeed have these permissions.

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/e2e/service_bootstrap.py [hunk 1]
```diff
@@ -87,21 +87,49 @@ def create_subnet(ec2, vpc_id: str, az: str, cidr: str) -> str:
...
     region = identity.get_region()
     ec2 = boto3.client("ec2", region_name=region)
+    iam = boto3.client("iam", region_name=region)
     # only normal zones, no localzones
     azs = map(lambda zone: zone['ZoneName'],
             filter(lambda zone: zone['OptInStatus'] == 'opt-in-not-required', ec2.describe_availability_zones()['AvailabilityZones']))
 
     vpc_id = create_vpc(ec2)
     subnets = [create_subnet(ec2, vpc_id, az, cidr) for (cidr,az) in zip(VPC_SUBNET_CIDR_BLOCK, azs)]
+    slr_name = create_service_linked_role(iam)
```
Comment (RedbackThomson) [inline test/e2e/service_bootstrap.py:127]: Do you ever use this role in the tests?

**Review by RedbackThomson — APPROVED** [review]

File: test/e2e/service_bootstrap.py [hunk 2]
```diff
@@ -87,21 +87,49 @@ def create_subnet(ec2, vpc_id: str, az: str, cidr: str) -> str:
     return subnet_id
 
 
+def create_service_linked_role(iam) -> str:
+    # TODO(jaypipes): There does not seem to be any way at all to list or
```
Comment (vijtrip2) [inline test/e2e/service_bootstrap.py:91]: From my previous experience of working on APIGateway team, SLR is created and managed by AWS services. Customers have nothing to do with it.
With that knowledge, I am wondering why do we need to create SLR ourselves.

API Gatewayv2 has the code path which created SLR in customer account and then used it to manage their resources.

**Review by vijtrip2 — COMMENTED** [review]
@jaypipes, What was the AWS API error that you saw in the controller logs during the e2e failure?

File: test/e2e/service_bootstrap.py (hunk 2 above)
Comment (jaypipes) [reply test/e2e/service_bootstrap.py:91]: @vijtrip2 unfortunately, for ElasticsearchService, that is not the case...

If you don't set up the SLR yourself and associate your AWS account with the `es.amazonaws.com` SLR, the following happens and ends up in a TerminalCondition:

> Before you can proceed, you must enable a service-linked role to give Amazon ES permissions to access your VPC.

Apparently, this isn't a new problem :(

https://github.com/cloudposse/terraform-aws-elasticsearch/issues/5

Note that Terraform issue was filed 3 years ago.

File: test/e2e/service_bootstrap.py (hunk 1 above)
Comment (jaypipes) [reply test/e2e/service_bootstrap.py:127]: See my comment above to @vijtrip2. If this isn't done, the e2e test that creates an ElasticsearchDomain using a private VPC does not work and the CR gets a TerminalCondition with this text in it:

> Before you can proceed, you must enable a service-linked role to give Amazon ES permissions to access your VPC.

**Pull request merged**